src/v3/src/util/idxUtils.ts

```diff
@@ -351,3 +352,19 @@ export const isVerifyFlow = (transaction: IdxTransaction): boolean => {
   const { context: { currentAuthenticatorEnrollment } } = transaction;
   return typeof currentAuthenticatorEnrollment !== 'undefined';
 };
+
+export const shouldSkipAfterRender = (idxTransaction: IdxTransaction): boolean => {
+  const {
+    nextStep: { name: stepName } = {},
+    neededToProceed,
+    messages,
+    context,
+  } = idxTransaction;
+  // When a transaction is in this scenario, it usually indicates a successful
+  // interaction code flow token response which means we should not trigger the afterRender event.
+  return idxTransaction.status === IdxStatus.SUCCESS
+    && stepName === 'cancel'
+    && neededToProceed.length === 0
+    && !messages?.length
+    && !context.messages;
```

Comment:
This all feels a bit smelly -- we're carving out a very specific set of cases to skip this hook. Is it necessarily bad to run the hook in this scenario?

Reply:
Oh yea, I agree with the smelly part (but wondering if there is another place that can handle this), the problem is that we are triggering an event that isn't really representative of anything on the UI. The event that is triggered is like this `{ controller: null, form: 'cancel' }`, so if a customer consumes this and relies on the afterRender event to do things or gain information, this could negatively impact them or at least confuse them since this is triggered right before an interaction code flow success event (which provides tokens). Also Gen 2 does not register this event.

Reply:
Also the flakiness is caused by the success event/tokens triggering before this erroneous event. So for example, sometimes the tokens success event will happen first then this afterRender cancel event, other times it is reversed. This is why avoiding triggering this event fixes the flakiness.

Reply:
Tests are passing with the above change, I was mostly concerned about interaction code flow where it could render view elements but I saw the code for that implementation and view elements will get added to `uischema` so we are safe. 🙏